src/utils.js

```diff
@@ -39,43 +38,37 @@ var getCommentFormatter = function (file) {
 
   switch (extension) {
     case 'css':
-      debug(logCb('css commentFormatter'));
+      debug(function() { return 'css commentFormatter';});
       commentFormatter = function(url) {
         return newline + "/*# sourceMappingURL=" + url + " */" + newline;
       };
       break;
     case 'js':
-      debug(logCb('js commentFormatter'));
+      debug(function() { return 'js commentFormatter'; });
       commentFormatter = function(url) {
         return newline + "//# sourceMappingURL=" + url + newline;
       };
       break;
     default:
-      debug(logCb('unknown commentFormatter'));
+      debug(function() { return 'unknown commentFormatter'; });
   }
 
   return commentFormatter;
 };
 
 var getInlinePreExisting = function(fileContent){
   if(sourceMapUrlRegEx().test(fileContent)){
-    debug(logCb('has preExisting'));
+    debug(function() { return 'has preExisting'; });
```

Comment:
Less memory pressure as the string does not get evaluated if that specific debug level is disabled. Without the closure the string is evaluated but possibly not logged. This adds up over time of lots of logging. I had a good link to read on this topic; but it is eluding me.